### PR TITLE
SPEC-1774 support AWS temp creds in CSFLE

### PR DESCRIPTION
**Pull request by kevinAlbs**
Adds integration tests for creating and passing temporary AWS credentials to authenticate KMS requests for CSFLE.

Drivers can obtain AWS temporary credentials with a new `set-temp-creds.sh` script included in drivers-evergreen-tools: https://github.com/mongodb-labs/drivers-evergreen-tools/pull/136 By default, the temporary credentials expire after 12 hours.

POC of the new tests running in Evergreen in libmongoc: https://github.com/mongodb/mongo-c-driver/pull/725

Tests require libmongocrypt 1.2.0: https://github.com/mongodb/libmongocrypt/releases/tag/1.2.0

### CODE REVIEW

File: source/client-side-encryption/tests/README.rst [hunk 1]
```diff
@@ -147,7 +147,32 @@ Then for each element in ``tests``:
 
 #. Create a **new** MongoClient using ``clientOptions``.
 
-   #. If ``autoEncryptOpts`` includes ``aws``, ``azure``, and/or ``gcp`` as a KMS provider, pass in credentials from the environment.
+   #. If ``autoEncryptOpts`` includes ``aws``, ``awsTemporary``, ``awsTemporaryNoSessionToken``, ``azure``, and/or ``gcp`` as a KMS provider, pass in credentials from the environment.
+      - ``awsTemporary``, and ``awsTemporaryNoSessionToken`` require temporary AWS credentials. These can be retrieved using the csfle `set-temp-creds.sh <https://github.com/mongodb-labs/drivers-evergreen-tools/tree/master/.evergreen/csfle>`_ script.
```
Comment (kevinAlbs) [inline source/client-side-encryption/tests/README.rst:151]: The rationale for testing `awsTemporaryNoSessionToken` is to ensure that drivers do not blindly pass the permanent AWS credentials. `awsTemporaryNoSessionToken` expects an error due to the missing session token.

**Review by rozza — APPROVED** [review]
LGTM

File: source/client-side-encryption/tests/README.rst (hunk 1 above)
Comment (saghm) [reply source/client-side-encryption/tests/README.rst:151]: These aren't just being added for test purposes, correct? Users will sometimes be using these options as well?

Comment (kevinAlbs) [reply source/client-side-encryption/tests/README.rst]: The `awsTemporarySessionToken` and `awsTemporaryNoSessionToken` keys are test-only. But the `aws.sessionToken` is a new option users can pass. That's noted in the spec itself.

The wonky names are arbitrary, and act as placeholders.

If a test specifies:
```
{
   "awsTemporarySessionToken" : {}
}
```

A test runner will configure the KMS providers as:
```
{
    "aws": {
        "accessKeyId": "...",
        "secretAccessKey": "...",
        "sessionToken": "..."
    }
}
```

I added some clarification on how to substitute test values, so hopefully this is clearer to implementers.

Comment (saghm) [reply source/client-side-encryption/tests/README.rst]: Oh, sorry, I completely missed that this was in the tests subdirectory; for some reason I thought this was adding a new test-only option for drivers to support, and was a little concerned about that. I have no issues with a test-only option of course though!

**Review by saghm — APPROVED** [review]

File: source/client-side-encryption/tests/README.rst (hunk 1 above)
Comment (kevinAlbs) [reply source/client-side-encryption/tests/README.rst]: Hah no worries, I think it was still worthwhile for making the test description clearer!

**Pull request merged**